### PR TITLE
Add documentation for strict mode

### PR DESCRIPTION
**Pull request by tpdorsey**
Adds documentation and examples for strict validation mode

## Change Summary

- New Strict Mode page
- Strict mode sections in Models, Model Config, and Fields
- Updated docstrings with references to Strict Mode usage docs

## Related issue number

Closes PYD-91

## Checklist

* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @dmontagu

### CODE REVIEW

**Comment by tpdorsey** [comment]
Please review

File: docs/usage/conversion_table.md [hunk 1]
```diff
@@ -1 +1,7 @@
+The following table provides details on how Pydantic converts data during validation in both strict and lax modes.
+
+"Lax mode" indicates the default behavior of Pydantic, where it attempts to coerce values to the correct type, when possible. "Strict mode" indicates the behavior of Pydantic when `strict=True` is set on either a field or `strict=True` is set on a model config.
```
Comment (Kludex) [inline docs/usage/conversion_table.md:3]:
```suggestion
"Lax mode" indicates the default behavior of Pydantic, where it attempts to coerce values to the correct type, when possible. "Strict mode" indicates the behavior of Pydantic when `strict=True` is set on either a field or on a model config.
```

File: docs/usage/models.md [hunk 2]
```diff
@@ -1199,7 +1199,38 @@ This is a deliberate decision of Pydantic, and in general it's the most useful a
 [here](https://github.com/pydantic/pydantic/issues/578) for a longer discussion on the subject.
 
 Nevertheless, [strict type checking](types/standard_types.md#strict-types) is partially supported.
-[TODO: Discuss strict mode, but also retain the previous link to types/standard_types.md#strict-types]
+
+## Strict Mode
```
Comment (Kludex) [inline docs/usage/models.md:1203]: I think this is the perfect place to talk about the "Strict Mode".

File: docs/usage/strict_mode.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
+There are some situations where you want Pydantic to throw an error instead of coercing data.
```
Comment (Kludex) [inline docs/usage/strict_mode.md:1]: I don't think this file will fit well in the current organization. The section on the `models.md` about the Strict Mode that you added in this PR is great already.

**Review by Kludex — CHANGES_REQUESTED** [review]

**Comment by Kludex** [comment]
The Strict Types are being explained on the `number_types.md` file. That may not be the best place for it.

Shall we include it with the "Strict Mode", or create a `strict_types.md` on the `Types` files?

**Comment by tpdorsey** [comment]
@Kludex revised per your comments, please re-check

**Review by Kludex — APPROVED** [review]
> Shall we include it with the "Strict Mode", or create a strict_types.md on the Types files?

So, no? 🤔

**Comment by tpdorsey** [comment]
@Kludex I think strict types is its own thing. I'll submit a separate PR for that.

**Pull request merged**